share/doc/wake/logging.md

```diff
@@ -0,0 +1,111 @@
+zero or more of `wake`'s own output streams, possibly with additional formatting applied.
+The wake codebase and standard libraries often refer to a _logger_ as a `logLevel`.
+
+What's the difference between the different loggers?
+Which of wake's own output streams the results are sent to.
+This can influence what you see on your console when you run wake.
+The logger, in combination with the command line flags you invoke wake with,
+controls *what* you see. The logger also controls what it looks like (what color it is).
+It also controls which *actual* stream it is going to coming out of the executing `wake` process
+(stderr or stdout, or other file descriptors 3-5).
+
+`wake` has a number of built in loggers in wake which all start with `log...`.
+You can also make your own logger using `mkLogLevel`.
+You can use this to customize the color you want it to be shown as (if at all),
+and giving more fine-grained control over which wake output stream it should go to.
+
+For stdout, the following table should be read top to bottom considering the command line arguments you've passed to wake.
+If you match a line in the table, stop!
+This is what will appear on the console as you watch wake execute, and what is sent to wake's stdout:
+
+|Flag               | logDebug | logInfo | logEcho | logReport | logWarning | logError | logNever | mkLogLevel "foo" Green|
+|-------------------|----------|---------|---------|-----------|------------|----------|----------|----------------------
+|Color on Console   |  Blue    |  Dim    |  default| Magenta   |  Yellow    |   Red    |          |      Green      |          
+|--stdout="foo"     |          |         |         |           |            |          |          |         x       |
+|--stdout="foo,info" |         |    x    |         |           |            |          |          |         x       |
+|--debug            |      x   |    x    |    x    |     x     |      x     |    x     |          |                 |
+|--verbose          |          |    x    |    x    |     x     |      x     |    x     |          |                 |
+|--quiet            |          |         |         |           |            |    x     |          |                 |
+|--no-tty<sup>*</sup>|         |         |    x    |     x     |      x     |    x     |          |                 |
+|(default)          |          |         |         |     x     |      x     |    x     |          |                 |
+
+<sup>*</sup> `--no-tty` also prevents any coloring from being applied.
+
+
+Unless overridden with `--stderr=...` the *only* thing that ever appears on stderr coming out of wake is `logError`,
+and it always appears there.
+
+The additional command line arguments `--fd:3`, `--fd:4`, `--fd:5` can be used to output to file descriptors 3, 4, 5.
+One example shell command would be, for some wake code that had `mkLogLevel "foo"`:
+
+```
+wake --fd3="foo" mywakecode 3&>foo.txt
```

Comment:
```suggestion
wake --fd:3="foo" mywakecode 3>foo.txt
```

The CLI option has a colon in its name. The way you put the `&` in the bash command will cause it to parse in a very different way than you would expect. `foocmd 3&>bar` parses the same as `foocmd 3 &>bar`, and the latter means call `foocmd` with a single argument of `3` and redirect both stdout and stderr to a file named `foo.txt`. (`&>` is shorthand for redirect both stdout and stderr.)

`&` unfortunately has lots of very different meanings in bash depending on exactly where it is. Without any redirections, it means background the process. On the left-hand side of a `>` redirection, `&` means redirect both stdout and stderr to the thing on the right. On the right-hand side of the `>` redirection, if it is followed by a number, then it means redirect to the file descriptor with that number. `>&3` means redirect stdout to file descriptor 3; `>3` means redirect stdout to file named `3`.

In your example here, you don't want any `&` at all because you're only trying to redirect fd 3 to a normally named file.